### DISCUSSION TITLE
my vps has ipv6, my internet providers only use ipv4, can i have ipv6 nat on my network?

### DISCUSSION THREAD

**Discussion by icolonder** [Ysurac/openmptcprouter#2689] [Q&A]
my vps has ipv6, my internet providers only use ipv4, can i have ipv6 nat on my network local?

**Comment by iambenmitchell** [comment]
IPv6 shouldn't use NAT. There's so many addresses that each device can have its own IP. Your VPS should have a /48 address. 

But to answer your question, yes. Your home will connect to your VPS over IPv4 and your VPS will provide your home with an IPv6 address block as well as its public IPv4 address.

**Reply by icolonder** [reply]
So if my vps assigns me a unique ipv6, not ipv6/48, I won't be able to navigate to pages with ipv6, is that correct?

**Reply by iambenmitchell** [reply]
You should be able to. 

If you have a Mac connected to your network you can run 

curl -6 icanhazip.com 

And see if it gives you an IPv6 address. If so, this if your public IPv6 address. 

You can also try various other IPv6 testing sites like https://test-ipv6.com

**Reply by icolonder** [reply]
I still haven't thanked you for your answers @iambenmitchell . Thank you very much.

My vps has ipv6/64 To get ipv6 connectivity on my local machine, this uses nat66 or proxy via SLAAC

And how do I activate it?
<img width="642" alt="Screenshot 2022-12-03 at 4 26 39 p m" src="https://user-images.githubusercontent.com/115506043/205464682-2505a70a-09cc-4908-9c50-ceeb8b3546a2.png">

**Reply by Ysurac** [reply]
You enable IPv6, then you will have ULA address (fdxx) on you local network.

**Comment by icolonder** [comment]
It doesn't work, I was hoping to get something from ntpv6, but it doesn't.

**Reply by Ysurac** [reply]
ntpv6 ?
Check on a client computer with tcpdump if you see IPv6 router advertisement, if not check LAN interface settings.

**Reply by icolonder** [reply]
My modem does not have ipv6, I was hoping to have ipv6 at home using the vps.

<img width="518" alt="Screenshot 2022-12-09 at 2 55 28 p m" src="https://user-images.githubusercontent.com/115506043/206794892-418742b2-3fcb-4dab-b979-c98e135df36b.png">

**Reply by Ysurac** [reply]
First check that your VPS really have IPv6 configured with for example ```curl -6 ifconfig.co```.
IPv6 must be configured on VPS before launching VPS install script.
If all is ok, check that IPv6 is really enabled in wizard, if yes and it's not working: reboot.

**Reply by icolonder** [reply]
I get my ipv6 public with curl. If I activate ipv6 in openmptcprouter it asks me for the ipv6 of my ISP. But I only have ipv4. That's why I was thinking of ntpv6

**Reply by Ysurac** [reply]
I don't know what you mean with ntpv6.
OpenMPTCProuter doesn't ask for your VPS IPv6, it ask for prefix if you have a /56 or /48 on the VPS, but it's not your case so you have to keep the ULA fdxx IPv6 prefix.
